sda-commons-server-spring-data-mongo/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,108 @@
+# Migration Guide
+
+## Getting started
+The spring-data-mongo package is available at the following coordinates:
+```groovy
+implementation "org.sdase.commons:sda-commons-server-spring-data-mongo"
+```
+
+Replacing the MorphiaBundle:
+
+```diff
+-  private final MorphiaBundle<AppConfiguration> morphiaBundle =
+-    MorphiaBundle.builder()
+-      .withConfigurationProvider(AppConfiguration::getMongo)
+-      .withEntityScanPackageClass(ConsentConfigurationEntity.class)
+-      .withValidation()
+-      .build();
+
++  private final SpringDataMongoBundle<AppConfiguration> mongoBundle = 
++    SpringDataMongoBundle.builder()
++      .withConfigurationProvider(AppConfiguration::getMongo)
++      .build();
```

Comment:
I remember that there are some case, that require to register the entities directly. Not sure if that was about inheritance or generics or if we already found a way that it is not needed.

Reply:
Wasn't it only about performance? I think the initialization will be done if you save your first entity. If you don't want that you can register your entity when initializing your bundle.

Reply:
In case you store an abstract class or interface - effectively multiple implementations - as entity or field like in [this test app](https://github.com/SDA-SE/sda-dropwizard-commons/blob/feat%2Fv3/sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/inheritance/InheritanceMappingITest.java#L122) it is needed. Two of the tests fail if you remove the entity registration. If your first action is a read action or you never stored the type you read in the currently running instance before reading, it will fail because the type is not known.